src/zosAccessor.ts

```diff
@@ -750,17 +750,32 @@ class ZosAccessor {
     public async getRCFromJESMSGLG(queryOption: JobLogOption): Promise<number | string> {
         return this.getJobLog(queryOption).then((log: string | SpoolFile[]) => {
             let rc;
+            let retcode = '';
             const logContents = log as string;
             if (logContents) {
-                const EYE_CATCHER = 'ENDED - RC=';
+                const EYE_CATCHER_1 = 'ENDED - RC=';
+                const EYE_CATCHER_2 = 'ENDED - ABEND=';
+                const EYE_CATCHER_3 = 'NOT AUTHORIZED';
                 logContents.split('\n').forEach((line: string) => {
-                    const posn = line.indexOf(EYE_CATCHER);
-                    if (posn !== -1) {
-                        rc = line.substring(posn + EYE_CATCHER.length).trim();
+                    const posn1 = line.indexOf(EYE_CATCHER_1);
+                    const posn2 = line.indexOf(EYE_CATCHER_2);
+                    const posn3 = line.indexOf(EYE_CATCHER_3);
+                   if (posn1 !== -1) {
+                        rc = parseInt(line.substring(posn1 + EYE_CATCHER_1.length).trim(), 10);
+                        retcode = 'RC ' + line.substring(posn1 + EYE_CATCHER_1.length).trim();
+                        }
+                    else if(posn2 !== -1) {
+                        retcode = 'ABEND ' + line.substring(posn2 + EYE_CATCHER_2.length).trim();
+                        rc = retcode;
+                        }
```

Comment:
As we talked, we don't mean to have the exact same error string as z/OSMF. So no need to parse the string from my perspective.

Reply:
We have another issue: can't return rc when job canceled. This part is for resolving the issue. The canceled job will have ABEND S222 code in JESMSG. The next part is for SEC ERROR, the reason is similar to canceled job.